packages/web-components/src/components/va-file-input/va-file-input.tsx

```diff
@@ -0,0 +1,168 @@
+import { 
+  Component, 
+  Element, 
+  Host, 
+  h, 
+  Prop, 
+  Build, 
+  forceUpdate,
+  Fragment,
+  Event,
+  EventEmitter
+} from '@stencil/core';
+import i18next from 'i18next';
+
+if (Build.isTesting) {
+  // Make i18next.t() return the key instead of the value
+  i18next.init({ lng: 'cimode' });
+}
+
+@Component({
+  tag: 'va-file-input',
+  styleUrl: 'va-file-input.css',
+  shadow: true,
+})
+
+export class VaFileInput {
+  @Element() el: HTMLElement;
+
+  /**
+   * The label for the button input.
+   */
+  @Prop() label?: string;
+
+  /**
+   * The name for the input element.
+   */
+  @Prop() name?: string;
+
+  /**
+   * The text displayed on the button
+   */
+  @Prop() buttonText: string;
+
+  /**
+   * Set the input to required and render the (Required) text.
+   */
+  @Prop() required?: boolean = false;
+
+  /**
+   * Allow the input to accept multiple files.
+   */
+  @Prop() multiple?: boolean = false;
```

Comment:
Should this prop change the user-facing text at all? Like `Upload file(s)`?

Reply:
I don't know the answer to this. @GnatalieH Do you know if we want to pluralize the button text when the component allows for multiple files? I did not see this in the original React component. Looking at [USWDS](https://designsystem.digital.gov/components/file-input/), they have two different types of text in the examples but no explicit guidance on pluralizing.

Reply:
@bkjohnson On further thought here, from a technical perspective, would we even want to control the text displayed within the component itself based on another prop? Shouldn't a pluralization be something the developer implementing it be responsible for updating (by changing the `buttonText` prop value itself)? Otherwise, we'd probably need to add an entirely new prop dedicated to pluralization text and I'm just not sure that's really necessary. This seems more like design guidelines than something we need to restrict within the component itself.  cc @GnatalieH

Reply:
@jamigibbs we won't have an instance where bulk file uploading will be an option, in fact we are discouraging that pattern because it's not mobile-friendly, and lots of errors can occur. So I don't think we need to worry about making the text plural. CC @bkjohnson

Reply:
It's unclear to me what the `buttonText` decision was. I see that there are lots of resolved comments which I haven't clicked on and opened, but [the original issue](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/966#issuecomment-1206886490) mentioned that a decision would be made:

> Button text (whether to lock it for editing or not): we will have a decision from Danielle and Beth next week about this

And I haven't seen anything else about that called out in the code or the PR description. So, if we're going to internalize that text, I think pluralizing it could be a reasonable thing to do.

---

>  we won't have an instance where bulk file uploading will be an option

Does this mean that we don't need `multiple` at all?

Reply:
@bkjohnson The "button text decision" was about if we should allow the text to be dynamic or a fixed static value (removing the prop). The decision was to allow it to be dynamic and default to "File upload". The conversation happened in [Slack](https://dsva.slack.com/archives/C01K37HRUAH/p1660240015362249?thread_ts=1659547593.525989&cid=C01K37HRUAH) so I'll update the ticket itself with that info.

Reply:
@bkjohnson and @jamigibbs the button text should default to "Upload file" rather than "File upload". The Content team recommends starting button text with a verb. Thanks.